samples/csharp_dotnetcore/13.core-bot/CognitiveModels/FlightBookingEx.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+
+namespace Microsoft.BotBuilderSamples
+{
+    // Extends the partial FlightBooking class with methods and properties that simplify accessing entities in the luis results
+    public partial class FlightBooking
+    {
+        // Gets the value of the From Entity and From Airport if present
+        // In some cases LUIS will recognize the From entity as a valid city, but if the CompositeEntity
+        // is not trained it will not recognize the city as a valid Airport.
+        public (string From, string Airport) FromEntities
+        {
+            get
+            {
+                var fromValue = Entities?._instance?.From?.FirstOrDefault()?.Text;
+                var fromAirportValue = Entities?.From?.FirstOrDefault()?.Airport?.FirstOrDefault()?.FirstOrDefault();
```

Comment:
This code need to be synced with the logic in corebot_updates
i.e. LUIS can find the composite entity because of the structure of the sentence but then miss the entity value because it doesn't match one of the closed list values.